src/daos/Canvas/client.ts

```diff
@@ -1,42 +1,46 @@
-import fetch from 'node-fetch';
+import fetch from "node-fetch";
 
 export interface IClanvasClient {
-  get: (path: string) => any | null;
+  get: (path: string) => Promise<any | null>;
 }
 
 class CanvasClient {
-  
-  public options: any;
-  public headers: any;
-
-  constructor(opts: any) {
-    if(!opts.token) { throw new Error("missing canvas api token option") }
-    this.options =  Object.assign({
-      base_url: 'https://lambdaschool.instructure.com/api/v1/',
-      ver: "/v1"
-    }, opts);
+  public options: Record<string, unknown>;
+  public headers: Record<string, unknown>;
+
+  constructor(opts: Record<string, unknown> = {}) {
+    if (!opts.token) {
+      throw new Error("missing canvas api token option");
+    }
+    this.options = Object.assign(
+      {
+        base_url: "https://lambdaschool.instructure.com/api/v1/",
+        ver: "/v1",
+      },
+      opts
+    );
     this.headers = {
-      "Authorization": `Bearer ${opts.token}`,
+      Authorization: `Bearer ${opts.token}`,
       "Content-Type": "application/vnd.api+json",
-      "Accept": "application/vnd.api+json"
-    }
+      Accept: "application/vnd.api+json",
+    };
   }
 
-  public async request(path: string = "", options = {}) {
-    const url = this.options.base_url + path
+  public async request(path = "", options: unknown = {}): Promise<any> {
+    const url = `${this.options.base_url}${path}`;
 
     const config = {
       ...options,
```

Comment:
```suggestion
  public async request(path = "", options: Record<string, unknown> = {}): Promise<any> {
    const url = `${this.options.base_url}${path}`;

    const config = {
      ...options as {},
```

Reply:
ok, Record has other issues as well. I'm redoing this to minimize type issues and the number of "any"s